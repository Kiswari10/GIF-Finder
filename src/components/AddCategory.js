import React, { useState } from "react";
import styled from "styled-components";
import { IconContext } from "react-icons";
import { AiOutlineSearch } from "react-icons/ai";

const Form = styled.form`
  width: 100%;
  height: 55px;
  background-color: red;
  padding: 0;
  display: flex;
  flex-direction: row;
  input {
    width: 92%;
    height: 100%;
    outline: none;
    border: none;
    margin: 0;
    padding: 0 1%;
    font-size: 18px;
    font-weight: 600;
  }
  button {
    background-image: linear-gradient(45deg, #16bf69, #22c4dc 51%, #16bf69);
    height: 100%;
    width: 8%;
    border: none;
    outline: none;
    cursor: pointer;
    margin: 0;
    padding: 0;
  }
`;

export const AddCategory = ( ) => {
  const [inputValue, setInputValue] = useState("");
  const handleInputValue = (event) => {
    setInputValue(event.target.value);
  };
  const handleSubmit = (e) => {
    e.preventDefault();
    console.log("OK");
  };
  return (
    <>
      <Form onSubmit={handleSubmit}>
        <input
          placeholder="Search GIFs"
          value={inputValue}
          onChange={handleInputValue}
        />
        <button onClick={handleSubmit}>
          <IconContext.Provider
            value={{
              size: "60%",
              color: "white",
              fontWeight: "bold",
            }}
          >
            <AiOutlineSearch />
          </IconContext.Provider>
        </button>
      </Form>
    </>
  );
};
