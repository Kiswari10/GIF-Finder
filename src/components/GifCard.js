import React from "react";
import PropTypes from "prop-types";
import { Card } from "react-bootstrap";

export const GifCard = ({ id, title, url }) => {
  return (
    <Card className="animate__animated animate__fadeIn">
      <Card.Img src={url} alt={title} />
    </Card>
  );
};

GifCard.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};
