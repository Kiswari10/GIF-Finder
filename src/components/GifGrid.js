import React, { useEffect, useState } from "react";
import { GifCard } from "./GifCard";
import { CardColumns } from "react-bootstrap";

export const GifGrid = ({ category }) => {
  const [gifs, setGifs] = useState([]);
  useEffect(() => {
    const getGifs = async () => {
      const response = await fetch(
        `http://api.giphy.com/v1/gifs/search?q=${category}&limit=12&api_key=PIJ9beNH3hdjavbtzJ5Nje2FdpV8cp0D`
      );
      const { data } = await response.json();
      const _gifs = data.map((img) => ({
        id: img.id,
        title: img.title,
        url: img.images.downsized_medium.url,
      }));
      setGifs(_gifs);
      console.log(_gifs);
    };
    getGifs();
  }, [category]);
  return (
    <div>
      <h3 className="logo">{category}</h3>
      <CardColumns>
        {gifs.map((gif) => (
          <GifCard {...gif} key={gif.id} />
        ))}
      </CardColumns>
    </div>
  );
};
