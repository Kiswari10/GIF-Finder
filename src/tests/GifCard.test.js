import React from "react";
import { shallow } from "enzyme";

import { GifCard } from "../components/GifCard";

describe("Test en <GifCard />", () => {
  const title = "Goku";
  const url = "https://localhos.pe";
  const wraper = shallow(<GifCard title={title} url={url} />);

  test("debe retornar el componente correctamente", () => {
    expect(wraper).toMatchSnapshot();
  });

  test("debe retornar la imagen con la url de los props", () => {
    const img = wraper.find("CardImg");
    expect(img.props().src).toBe(url);
  });

  test("debe retornar la imagen con el alt de los props", () => {
    const img = wraper.find("CardImg");
    expect(img.prop("alt")).toBe(title);
  });

  test("debe retornar la clase animate__fadeIn", () => {
    const div = wraper.find("Card");
    const className = div.prop("className");
    expect(className.includes("animate__fadeIn")).toBe(true);
  });
});
