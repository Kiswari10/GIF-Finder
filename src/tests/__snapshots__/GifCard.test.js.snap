// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test en <GifCard /> debe retornar el componente correctamente 1`] = `
<Card
  body={false}
  className="animate__animated animate__fadeIn"
>
  <CardImg
    alt="Goku"
    src="https://localhos.pe"
    variant={null}
  />
</Card>
`;
